src/sentry/event_manager.py

```diff
@@ -349,19 +349,25 @@ def clean(d):
 
         self._data = data
 
-    def normalize(self):
+    def normalize(self, project_id=None):
         with metrics.timer("events.store.normalize.duration"):
-            self._normalize_impl()
+            self._normalize_impl(project_id=project_id)
+
+    def _normalize_impl(self, project_id=None):
+        if self._project and project_id and project_id != self._project.id:
+            raise RuntimeError(
+                "Initialized EventManager with one project ID and called save() with another one"
```

Comment:
Can we prevent this by not initializing EventManager anymore?

Reply:
yes, but that's a large diff. let's make smaller steps